### PR TITLE
HADOOP-19249 : Getting NullPointerException when the unauthorised user tries to perform the key operation

### PR DESCRIPTION
**Pull request by dhavalshah9131**
…r tries to perform the key operation

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   6m 53s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  36m 15s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  10m 19s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |   9m 13s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 53s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 24s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  22m 35s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 29s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |  10m 29s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 28s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |   9m 28s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 59s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 36s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 32s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 38s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  24m 38s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 35s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   | 157m  1s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6984 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 5fbe75e12949 5.15.0-117-generic #127-Ubuntu SMP Fri Jul 5 20:13:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 8d897fd4e45bf7260fad1df1881dc8bda82cbe30 |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/1/testReport/ |
| Max. process+thread count | 1285 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 package org.apache.hadoop.crypto.key.kms;
 
+import io.netty.util.internal.StringUtil;
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java]: -1 to any netty. use org.apache.commons.lang3.StringUtils

**Review by steveloughran — CHANGES_REQUESTED** [review]
need to switch to the commons lang version, as it is broadly used across all client deployments

now, can you add a test for this?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 21s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  36m  3s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   9m  1s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |   8m 14s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   0m 44s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m  3s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 47s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 29s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 25s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 32s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 41s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |   8m 41s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m  9s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |   8m  9s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 37s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 32s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 28s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 30s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 141m  7s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6984 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 7e19d20b37a9 5.15.0-117-generic #127-Ubuntu SMP Fri Jul 5 20:13:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 77ed6fc274230475bd7c4dda21d9499f7b8f1fcb |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/2/testReport/ |
| Max. process+thread count | 1282 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
  */
 package org.apache.hadoop.crypto.key.kms;
 
-import io.netty.util.internal.StringUtil;
+import org.apache.commons.lang3.StringUtils;
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java]: move this one line down to try and keep imports in order. its a losing battle. but we try...

**Review by steveloughran — COMMENTED** [review]
+1 pending the import reorder

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java (hunk 2 above)
Comment (dhavalshah9131) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java]: Done.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 19s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m 36s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   9m  3s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |   8m 10s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   0m 43s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 56s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 48s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 36s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 31s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 20s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 31s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |   8m 31s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 17s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |   8m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 58s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   1m 35s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 46s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 36s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 137m 50s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6984 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux aa2cc8a8fca5 5.15.0-117-generic #127-Ubuntu SMP Fri Jul 5 20:13:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 738c42ee1d1f654d12a3075a4e2771de2b5f2163 |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/3/testReport/ |
| Max. process+thread count | 1274 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6984/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by steveloughran — APPROVED** [review]

**Pull request merged**